src/libs/ajax/DatasetBuilder.ts

```diff
@@ -1,11 +1,60 @@
+import { DomainType, ProgramDataType } from 'src/pages/library/datasetBuilder/dataset-builder-types';
+
 export interface DatasetResponse {
   name: string;
+  programDataTypes: ProgramDataType[];
+  domainTypes: DomainType[];
 }
 
 export interface DatasetBuilderContract {
   retrieveDataset: (datasetId: string) => Promise<DatasetResponse>;
 }
 
-export const DatasetBuilder = (): DatasetBuilderContract => ({
-  retrieveDataset: (datasetId) => Promise.resolve({ name: 'AnalytiXIN', id: datasetId }),
+export const DatasetBuilder = (_signal?: AbortSignal): DatasetBuilderContract => ({
+  retrieveDataset: (_datasetId) =>
+    Promise.resolve({
+      name: 'AnalytixIndiana',
+      programDataTypes: [
+        { id: 1, name: 'Year of birth', dataType: 'range', min: 1900, max: 2023 },
+        {
+          id: 2,
+          name: 'Ethnicity',
+          dataType: 'list',
+          values: [
+            { name: 'Hispanic or Latino', id: 20 },
+            { name: 'Not Hispanic or Latino', id: 21 },
+            { name: 'No Matching Concept', id: 0 },
+          ],
+        },
+        {
+          id: 3,
+          name: 'Gender identity',
+          dataType: 'list',
+          values: [
+            { name: 'FEMALE', id: 22 },
+            { name: 'MALE', id: 23 },
+            { name: 'No Matching Concept', id: 0 },
+          ],
+        },
+        {
+          id: 4,
+          name: 'Race',
+          dataType: 'list',
+          values: [
+            { name: 'American Indian or Alaska Native', id: 24 },
+            { name: 'Asian', id: 25 },
+            { name: 'Black', id: 26 },
+            { name: 'White', id: 27 },
+            { name: 'No Matching Concept', id: 0 },
+          ],
+        },
+      ],
+      domainTypes: [
+        { id: 10, category: 'Condition', values: ['Heart Disease', 'Diabetes', 'Cancer'] },
```

Comment:
TOL (no change required here): I think this works here for this stubbing, but thinking about eventual API's, we may want values to be a separate API call just for size of response reasons?